### ISSUE TITLE
ErrorException: Warning: implode(): Invalid arguments passed

### ISSUE THREAD

**Issue by zetxek** [Paazl/magento2-checkout-widget#52]
An issue found in our production environment, while checking the logs:

```
ErrorException: Warning: implode(): Invalid arguments passed
#34 /deployment/releases/20210122102939/vendor/paazl/magento2-checkout-widget/Model/Api/Builder/Order.php(257): Paazl\CheckoutWidget\Model\Api\Builder\Order::parseAddress
#33 /deployment/releases/20210122102939/vendor/paazl/magento2-checkout-widget/Model/Api/Builder/Order.php(106): Paazl\CheckoutWidget\Model\Api\Builder\Order::getCreateOrderData
#32 /deployment/releases/20210122102939/vendor/paazl/magento2-checkout-widget/Model/Api/Processor/SendToService.php(107): Paazl\CheckoutWidget\Model\Api\Processor\SendToService::process
#31 /deployment/releases/20210122102939/vendor/paazl/magento2-checkout-widget/Cron/SendOrders.php(83): Paazl\CheckoutWidget\Cron\SendOrders::execute
#30 [internal](0): call_user_func_array
#29 /deployment/releases/20210122102939/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(350): Magento\Cron\Observer\ProcessCronQueueObserver::_runJob
#28 /deployment/releases/20210122102939/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(809): Magento\Cron\Observer\ProcessCronQueueObserver::tryRunJob
#27 /deployment/releases/20210122102939/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(776): Magento\Cron\Observer\ProcessCronQueueObserver::processPendingJobs
#26 /deployment/releases/20210122102939/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(267): Magento\Cron\Observer\ProcessCronQueueObserver::execute
#25 /deployment/releases/20210122102939/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Cron\Observer\ProcessCronQueueObserver\Interceptor::___callParent
#24 /deployment/releases/20210122102939/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Cron\Observer\ProcessCronQueueObserver\Interceptor::Magento\Framework\Interception\{closure}
#23 /deployment/releases/20210122102939/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Cron\Observer\ProcessCronQueueObserver\Interceptor::___callPlugins
#22 /Magento/Cron/Observer/ProcessCronQueueObserver/Interceptor.php(26): Magento\Cron\Observer\ProcessCronQueueObserver\Interceptor::execute
#21 /deployment/releases/20210122102939/vendor/magento/framework/Event/Invoker/InvokerDefault.php(88): Magento\Framework\Event\Invoker\InvokerDefault::_callObserverMethod
#20 /deployment/releases/20210122102939/vendor/magento/framework/Event/Invoker/InvokerDefault.php(74): Magento\Framework\Event\Invoker\InvokerDefault::dispatch
#19 /deployment/releases/20210122102939/vendor/magento/framework/Event/Manager.php(66): Magento\Framework\Event\Manager::dispatch
#18 /Magento/Framework/Event/Manager/Proxy.php(95): Magento\Framework\Event\Manager\Proxy::dispatch
#17 /deployment/releases/20210122102939/vendor/magento/framework/App/Cron.php(86): Magento\Framework\App\Cron::launch
#16 /deployment/releases/20210122102939/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\App\Cron\Interceptor::___callParent
#15 /deployment/releases/20210122102939/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\App\Cron\Interceptor::Magento\Framework\Interception\{closure}
#14 /deployment/releases/20210122102939/vendor/justbetter/magento2-sentry/Plugin/GlobalExceptionCatcher.php(58): JustBetter\Sentry\Plugin\GlobalExceptionCatcher::aroundLaunch
#13 /deployment/releases/20210122102939/vendor/magento/framework/Interception/Interceptor.php(135): Magento\Framework\App\Cron\Interceptor::Magento\Framework\Interception\{closure}
#12 /deployment/releases/20210122102939/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\App\Cron\Interceptor::___callPlugins
#11 /Magento/Framework/App/Cron/Interceptor.php(26): Magento\Framework\App\Cron\Interceptor::launch
#10 /deployment/releases/20210122102939/vendor/magento/module-cron/Console/Command/CronCommand.php(117): Magento\Cron\Console\Command\CronCommand::execute
#9 /deployment/releases/20210122102939/vendor/symfony/console/Command/Command.php(255): Symfony\Component\Console\Command\Command::run
#8 /deployment/releases/20210122102939/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Cron\Console\Command\CronCommand\Interceptor::___callParent
#7 /deployment/releases/20210122102939/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Cron\Console\Command\CronCommand\Interceptor::Magento\Framework\Interception\{closure}
#6 /deployment/releases/20210122102939/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Cron\Console\Command\CronCommand\Interceptor::___callPlugins
#5 /Magento/Cron/Console/Command/CronCommand/Interceptor.php(26): Magento\Cron\Console\Command\CronCommand\Interceptor::run
#4 /deployment/releases/20210122102939/vendor/symfony/console/Application.php(1005): Symfony\Component\Console\Application::doRunCommand
#3 /deployment/releases/20210122102939/vendor/symfony/console/Application.php(271): Symfony\Component\Console\Application::doRun
#2 /deployment/releases/20210122102939/vendor/magento/framework/Console/Cli.php(115): Magento\Framework\Console\Cli::doRun
#1 /deployment/releases/20210122102939/vendor/symfony/console/Application.php(147): Symfony\Component\Console\Application::run
#0 /deployment/releases/20210122102939/bin/magento(23): null
```

It seems that the address parser might be missing some error handling/data validation before doing operations on the strings.

**Comment by zetxek** [comment]
> Hi @zetxek ,
> Thank you for opening and reporting this issue. We have created an internal task for this issue and will try to come back to you as soon as possible. Thank you for your patience!

thanks! Also, please mind that I don't think it's a fatal error - but a Warning.

**Comment by Marvin-Magmodules** [comment]
Hi @zetxek,We are happy to share that we've just released the new [1.4.1](https://github.com/Paazl/magento2-checkout-widget/releases/tag/1.4.1) version with a fix for this issue. Thank you again for the detailed reports and patience. We are closing this issue now but please feel free to reopen the issue if this still occurs.
